location-service/src/main/java/com/movie/recommender/location/repository/CountryRepository.java

```diff
@@ -1,10 +1,16 @@
 package com.movie.recommender.location.repository;
 
-import com.movie.recommender.location.model.entity.Country;
+import com.movie.recommender.location.model.dto.CountryCityDTO;
 import org.springframework.data.jpa.repository.JpaRepository;
-
+import com.movie.recommender.location.model.entity.Country;
+import org.springframework.data.jpa.repository.Query;
 import java.util.Optional;
+import java.util.List;
 
 public interface CountryRepository extends JpaRepository<Country, Long> {
     Optional<Country> findByName(String name);
+    @Query("SELECT new com.movie.recommender.location.model.dto.CountryCityDTO(cn.name, ct.name) " +
```

Comment:
you don’t need to join with the locations table in this query, as it may cause duplicates if there are multiple users from the same city
to simplify and avoid duplicates, update the query to something like this
```
SELECT cn.name, ct.name
FROM countries cn
JOIN cities ct ON ct.country_id = cn.id
```